src/mongocxx/test/client_session.cpp

```diff
@@ -317,10 +317,10 @@ class session_test {
         apm_opts.on_command_started([&](const events::command_started_event& event) {
             std::string command_name{event.command_name().data()};
 
-            // Ignore auth commands like "saslStart", and handshakes with "isMaster".
+            // Ignore auth commands like "saslStart", and handshakes with "hello" (and the legacy "hello" command).
             std::string sasl{"sasl"};
             if (event.command_name().substr(0, sasl.size()).compare(sasl) == 0 ||
-                command_name.compare("isMaster") == 0) {
+                command_name.compare("hello") == 0 || command_name.compare("isMaster") == 0) {
```

Comment:
No need to change this, but I wanted to point out that although std::string::compare() is fine, if you only care about exact equality, it's less work to write (and a bit more familiar to read):

if(command_name == "hello") {}

The main difference-- and point of the compare() member function-- is that operator==() gives a Boolean result (that is. equal-to), whereas compare() gives lexicographic order: less-than and greater-than 0 are distinct.

(Further reference: https://www.cplusplus.com/reference/string/string/compare/)

Reply:
Ah, that's interesting.  I'm more familiar with string manipulation in Java and Python, so I just followed the existing pattern here.  However, I'll keep this in mind if there is a need to modify this in the future.